cluster.go

```diff
@@ -987,9 +988,30 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 		}
 	}
 
+	var trustedPeers []peer.ID
+	peers, err := c.consensus.Peers(ctx)
+	if err != nil {
+		return err
+	}
+
+	hashCache := make(utils.HashCache)
+
+	for _, p := range peers {
+		// We are also part of trusted peers and we must avoid a comparision
+		// with otherselves. Since a peer is not closer than itself, that
+		// comparision would result in false, making us leave unpinning for
+		// some other peer.
+		if p == c.id || !c.consensus.IsTrustedPeer(ctx, p) {
+			continue
+		}
+		trustedPeers = append(trustedPeers, p)
+	}
```

Comment:
A helper function to get a slice of trusted peers that does not include ourselves seems like a useful thing to have.